src/transformStateWithClones.js

```diff
@@ -59,13 +59,41 @@
  * {foo: 'bar', bar: 'foo'}.
  *
  *
- * @param {Object} state
+ * @param {Object} res
  * @param {Object[]} transforms
  *
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const res = [];
+  const copy = {
+    ...state,
+  };
+
+  for (let i = 0; i < transforms.length; i++) {
+    if (transforms[i].operation === 'addProperties') {
+      Object.assign(copy, transforms[i].properties);
+      res.push({ ...copy });
```

Comment:
you can push your object "copy" outside of "if-else" blocks then you avoid of duplicating your code